src/PortalGunHooks.js

```diff
@@ -0,0 +1,73 @@
+// @flow
+import * as React from 'react';
+
+class Store {
+  static id = 100000;
+
+  listeners = [];
+  idsToNodes = new Map();
+
+  addNode(node, id) {
+    this.idsToNodes.set(id, node);
+  }
+
+  deleteNode(id) {
+    this.idsToNodes.delete(id);
+    this.queueUpdate();
+  }
+
+  addListener(handler) {
+    this.listeners.push(handler);
+  }
+
+  removeListener(handler) {
+    this.listeners.splice(this.listeners.indexOf(handler), 1);
+  }
+
+  queueUpdate() {
+    setTimeout(() => this.listeners.forEach(listener => listener()), 0);
+  }
+}
+
+function SourceCmp({ children, store }) {
+  const [id] = React.useState(() => Store.id + 1);
```

Comment:
You need to increment Store.id, otherwise you always get `100001`. I'd probably make a separate private hook function for this.

```js
const useId = () => {
  const ref = React.useRef();
  if (!ref.current) {
    Store.id += 1;
    ref.current = Store.id;
  }
  return ref.current;
}
```

And then it's just `const id = useId()` to use it.